app/models/spree/order_decorator.rb

```diff
@@ -333,6 +333,18 @@ def pending_payments
     payments.select {|p| p.state == "checkout"} # Original definition
   end
 
+  # Although Spree 2 supports multi shipments per order, in OFN we keep the rule one shipment per order
+  #   Thus, this method sets the given shipping_method on the first and only shipment in the order
+  def shipping_method=(shipping_method = nil)
+    if shipments.empty?
+      shipment = Spree::Shipment.new
+      shipment.order = self
+      shipment.save
+      shipments << shipment
+    end
+    shipments.first.add_shipping_method(shipping_method, true)
+  end
```

Comment:
Have we investigated how this works with the way Spree creates shipments in https://github.com/openfoodfoundation/spree/blob/fe0a1311abb097bf3ac8001a24246c6cff5ecdbb/core/app/models/spree/order.rb#L506-L515 ? That's the only place I know in Spree's codebase where shipments are created.

Reply:
yes, I have seen that code. I am closing this PR actually, we need to fix the specs and then we will see if this is still required.